test/specs/debugApexTests.e2e.ts

```diff
@@ -0,0 +1,160 @@
+
+    // Open the Test Sidebar
+    const testingSideBarView = await testingView?.openView();
+    expect(testingSideBarView).toBeInstanceOf(SideBarView);
+
+    const sidebar = workbench.getSideBar();
+    const sidebarView = sidebar.getContent();
+    const apexTestsSection = await sidebarView.getSection('APEX TESTS');
+    expect(apexTestsSection.elem).toBePresent();
+
+    // Click the debug tests button that is shown to the right when you hover a test class name on the Test sidebar
+    const apexTestItem = await apexTestsSection.findItem('ExampleApexClass1Test') as TreeItem;
+    await apexTestItem.select();
+    const runTestsAction = await apexTestItem.getActionButton('Debug Tests');
+    await runTestsAction!.elem.click();
+    await utilities.pause(1);
+
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running Debug Test(s)', fiveMinutes);
+
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'Debug Test(s) successfully ran');
+    expect(successNotificationWasFound).toBe(true);
+
+    // Continue with the debug session
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+  });
+
+  step('Debug a Single Apex Test Method via the Test Sidebar', async () => {
+    const workbench = await browser.getWorkbench();
+    const testingView = await workbench.getActivityBar().getViewControl('Testing');
+
+    // Open the Test Sidebar
+    const testingSideBarView = await testingView?.openView();
+    expect(testingSideBarView).toBeInstanceOf(SideBarView);
+
+    const sidebar = workbench.getSideBar();
+    const sidebarView = sidebar.getContent();
+    const apexTestsSection = await sidebarView.getSection('APEX TESTS');
+    expect(apexTestsSection.elem).toBePresent();
+
+    // Hover a test name under one of the test class sections and click the debug button that is shown to the right of the test name on the Test sidebar
+    const apexTestItem = await apexTestsSection.findItem('validateSayHello') as TreeItem;
+    await apexTestItem.select();
+    const runTestAction = await apexTestItem.getActionButton('Debug Single Test');
+    await runTestAction!.elem.click();
+    await utilities.pause(1);
+
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running Debug Test(s)', fiveMinutes);
+
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'Debug Test(s) successfully ran');
+    expect(successNotificationWasFound).toBe(true);
+
+    // Continue with the debug session
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+  });
+
+  step('Tear down and clean up the testing environment', async () => {
+    await testSetup.tearDown();
+  });
+});
```

Comment:
The code uses break points.  Is it worth having tests which debug using check points as well?

Reply:
In the trailApexReplayDebugger suite there's already a test that uses check points, so I think it's ok like this